### PR TITLE
Addons on Text Input - Functionality / Docs / Tests

### PR DESCRIPTION
**Pull request by ayush547**
### Platforms Impacted
- [ ] iOS
- [ ] macOS
- [ ] win32 (Office)
- [ ] windows
- [x] android

### Description of changes

- Add E2E  setup and tests for input on Android. Expand jest tests.
- Add documentations for Input.
- Add support for changing icon to filled one when in focus and outline when in error.
- Add accessibility label to input and accessoryPressable. Make input come into focus when using keyboard to interact with accessoryPressable.
- Rename secondaryText to accessoryText. Disable accessoryText and accessoryIcon when text is empty.
- Add keyboard avoiding behavior on Input test page.
- Update tester samples to show textInputProps usage.

### Verification

| Light                                       | Dark                                      |
|----------------------------------------------|--------------------------------------------|
| input_light | input_dark |
| Placeholder State Light | Placeholder State Dark |
| input_placeholder_light | input_placeholder_dark |
| Focused State Light | Focused State Dark |
| input_focused_light | input_focused_dark |
| Typing State Light | Typing State Dark |
| input_typing_light |  input_typing_dark |
| Filled State Light | Filled State Dark |
| input_filled_light | input_filled_dark | 
| Error State Light | Error State Dark |
| input_error_light | input_error_dark |

### Pull request checklist

This PR has considered (when applicable):
- [x] Automated Tests
- [x] Documentation and examples
- [x] Keyboard Accessibility
- [x] Voiceover
- [ ] Internationalization and Right-to-left Layouts

### CODE REVIEW

File: packages/components/Input/SPEC.md
Comment (rohanprasadofficial) [inline packages/components/Input/SPEC.md:1]: I think we need to add the accessiblity behaviour as well.

File: packages/components/Input/SPEC.md [hunk 1]
```diff
@@ -2,4 +2,374 @@
...
+  /**
+   * Determines which keyboard to open.
+   * enum("default", 'numeric', 'email-address', "ascii-capable", 'numbers-and-punctuation', 'url', 'number-pad', 'phone-pad', 'name-phone-pad',
+   * 'decimal-pad', 'twitter', 'web-search', 'visible-password')
+   * Determines which keyboard to open, e.g.numeric.
+   * The following values work across platforms: - default - numeric - email-address - phone-pad
+   * The following values work on iOS: - ascii-capable - numbers-and-punctuation - url - number-pad - name-phone-pad - decimal-pad - twitter - web-search
+   * The following values work on Android: - visible-password
+   */
+  type?: KeyboardTypeOptions | undefined; // Uses KeyBoardType under the hood
```
Comment (rohanprasadofficial) [inline packages/components/Input/SPEC.md:195]: super nit: KeyboardType ? 🤔

File: apps/fluent-tester/src/TestComponents/Input/InputDefault.tsx
Comment (rohanprasadofficial) [inline apps/fluent-tester/src/TestComponents/Input/InputDefault.tsx:1]: Testing and adding a customized example would be great here.

File: packages/components/Input/src/Input.types.ts [hunk 2]
```diff
@@ -127,11 +127,33 @@ export interface InputTokens extends InputCoreTokens {
...
+   */
+  accessoryIconAccessibilityLabel?: string;
+
   /*
    * Source URL or name of the icon to show on the input.
+   *
+   * Based on fluent guidelines this icon should be an "outline" icon.
+   * FocusedStateIcon is applied onFocus and should be a "filled" icon.
    */
   icon?: IconSourcesType;
```
Comment (ankraj12) [inline packages/components/Input/src/Input.types.ts:146]: rename to `defaultIcon `to keep it inline with the comment below.

Comment (ankraj12) [reply packages/components/Input/src/Input.types.ts:146]: On similar - is icon an optional property?

File: packages/components/Input/src/__tests__/Input.test.tsx [hunk 3]
```diff
@@ -5,19 +5,53 @@ import * as renderer from 'react-test-renderer';
...
+    const tree = renderer.create(<Input placeholder="Test" icon={iconProps} />).toJSON();
+    expect(tree).toMatchSnapshot();
+  });
+
+  it('Input in error state', () => {
+    const tree = renderer.create(<Input error="Error" />).toJSON();
     expect(tree).toMatchSnapshot();
   });
 
   it('Input simple rendering does not invalidate styling', () => {
```
Comment (ankraj12) [inline packages/components/Input/src/__tests__/Input.test.tsx:46]: The tests well cover the core scenarios :)

**Review by ankraj12 — APPROVED** [review]
Approving with minor feedbacks

File: packages/components/Input/src/Input.types.ts (hunk 2 above)
Comment (ayush547) [reply packages/components/Input/src/Input.types.ts:146]: Alright renaming.
Yes, the icon is optional!

**Pull request merged**